ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -277,23 +277,31 @@ private static Statistics collectStatistics(HiveConf conf, PrunedPartitionList p
     boolean metaTable = table.getMetaTable() != null;
 
     if (!table.isPartitioned()) {
+      long ds, nr, fs;
+      if (table.isNonNative() && table.getStorageHandler().canProvideBasicStatistics()) {
+        Map<String, String> icebergBasicStatMap = table.getStorageHandler().getBasicStatistics(Partish.buildFor(table));
+        ds = Long.parseLong(icebergBasicStatMap.get("totalSize"));
+        nr = Long.parseLong(icebergBasicStatMap.get("numRows"));
+        fs = Long.parseLong(icebergBasicStatMap.get("numFiles"));
```

Comment:
Should a null check be added here and/or handle NumberFormatException?

What would be the stats values if it can not be extracted via storage handler?
In case of non-storage handler path there are so called `enhancers` to estimate and fix the values:
```
basicStatsFactory.addEnhancer(new BasicStats.RowNumEstimator(estimateRowSizeFromSchema(conf, schema)));
basicStatsFactory.addEnhancer(new BasicStats.SetMinRowNumber01());
```

Reply:
1. It seems to me that it check if the currentSnapshot exists or not but what if summary does not contain for example `SnapshotSummary.TOTAL_DATA_FILES_PROP`.

2. The line
```
Long.parseLong(icebergBasicStatMap.get("totalSize"))
```
can throw `NumberFormatException` because
* `icebergBasicStatMap.get("totalSize")` returns null because the map does have an entry with `totalSize`
*  or it has such entry but the value is not a valid `Long` value.

3. I think the name `icebergBasicStatMap` is too specific here because at this point we should not expect an `IcebergStorageHandler` but any `HiveStorageHandler` which supports providing BasicStatistics. So a name like `basicStatMap` would be more generic.

Reply:
Thanks @kasakrisz , i have updated the PR to catch numberFormatException and fall back to the pre existing implementation of basic stats.